packages/nx-monorepo/src/nx-monorepo.ts

```diff
@@ -239,6 +305,92 @@ export class NxMonorepoProject extends TypeScriptProject {
     });
   }
 
+  /**
+   * Helper to format `npx nx run-many ...` style command.
+   * @param options
+   */
+  public formatNxRunManyCommand(options: NxRunManyOptions): string {
+    const cmd: string[] = [
+      "npx nx run-many",
+      `--target=${options.target}`,
+      `--output-style=${options.outputStyle || "stream"}`,
+    ];
+    if (options.configuration) {
+      cmd.push(`--configuration=${options.configuration}`);
+    }
+    if (options.runner) {
+      cmd.push(`--runner=${options.runner}`);
+    }
+    if (options.parallel) {
+      cmd.push(`--parallel=${options.parallel}`);
+    }
+    if (options.skipCache) {
+      cmd.push("--skip-nx-cache");
+    }
+    if (options.ignoreCycles) {
+      cmd.push("--nx-ignore-cycles");
+    }
+    if (options.noBail !== true) {
+      cmd.push("--nx-bail");
+    }
+    if (options.projects && options.projects.length) {
+      cmd.push(`--projects=${options.projects.join(",")}`);
+    }
+    if (options.exclude) {
+      cmd.push(`--exclude=${options.exclude}`);
+    }
+    if (options.verbose) {
+      cmd.push("--verbose");
+    }
+
+    return cmd.join(" ");
+  }
+
+  /**
+   * Overrides "build" related project tasks (build, compile, test, etc.) with `npx nx run-many` format.
+   * @param task - The task or task name to override
+   * @param options - Nx run-many options
+   * @param force - Force unlocking task (eg: build task is locked)
+   * @returns - The task that was overridden
+   * @internal
+   */
+  protected _overrideNxBuildTask(
+    task: Task | string | undefined,
+    options: NxRunManyOptions,
+    force?: boolean
+  ): Task | undefined {
+    if (typeof task === "string") {
+      task = this.tasks.tryFind(task);
+    }
+
+    if (task == null) {
+      return;
+    }
+
+    if (force) {
+      // @ts-ignore - private property
+      task._locked = false;
+    }
+
+    task.reset(this.formatNxRunManyCommand(options), {
+      receiveArgs: true,
+    });
+
+    task.description += " for all affected projects";
+
+    return task;
+  }
+
+  /**
+   * Add project task that executes `npx nx run-many ...` style command.
+   */
+  public addNxRunManyTask(name: string, options: NxRunManyOptions): Task {
```

Comment:
I assume this is so clients can add additional tasks wrapped in nx?

Reply:
Yes that is correct